src/core/sys/posix/dirent.d

```diff
@@ -155,19 +157,38 @@ else version (FreeBSD)
         DT_WHT      = 14
     }
 
-    align(4)
-    struct dirent
+    static if (__FreeBSD_version >= 1200000)
     {
-        uint      d_fileno;
-        ushort    d_reclen;
-        ubyte     d_type;
-        ubyte     d_namlen;
-        char[256] d_name = 0;
+        struct dirent
+        {
+            ino_t     d_fileno;
+            off_t     d_off;
+            ushort    d_reclen;
+            ubyte     d_type;
+            ushort    d_namlen;
```

Comment:
There is a `__uint8_t  d_pad0;` before `d_namelen` in dirent.h on FreeBSD 12. So, there should be a `ubyte d_pad0;` above `d_namelen` here. If this was copied from what I did before, then I screwed it up.

https://github.com/freebsd/freebsd/blob/release/12.2.0/sys/sys/dirent.h#L71

Reply:
Yep, the content of the structs/aliases are all a copy-pasto from your PR.